sdks/python/conftest.py

```diff
@@ -0,0 +1,29 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Pytest configuration and custom hooks."""
+
+from __future__ import absolute_import
+
+import sys
+
+# See pytest.ini for main collection rules.
+collect_ignore_glob = []
+if sys.version_info < (3,):
+  collect_ignore_glob.append('*_py3.py')
+for minor in [5, 6, 7, 8, 9]:
+  if sys.version_info < (3, minor):
+    collect_ignore_glob.append('*_py3%d.py' % minor)
```

Comment:
How about something like this:

```python
MAX_SUPPORTED_PYTHON_VERSION = (3, 8)

# See pytest.ini for main collection rules.
collect_ignore_glob = []
if sys.version_info < (3,):
  collect_ignore_glob.append('*_py3.py')
else:
  for minor in range(sys.version_info.minor +1, MAX_SUPPORTED_PYTHON_VERSION[1] + 1):
    collect_ignore_glob.append('*_py3%d.py' % minor)
```

Reply:
oh, btw, what did you think of this suggestion?

Reply:
IDK. It's more verbose but I don't have a preference.

Ideally, I'd like a single source of truth for the list of Python supported versions (`python_requires` in setup.py is such a source but it's hard to parse).

Reply:
> IDK. It's more verbose but I don't have a preference.

I think it'll be easier to maintain long term: just update the max support python version.  

But maintainability aside,  it seems a bit odd to test for our lower bound in a loop when we know precisely what the lower bound is: `sys.version_info.minor +1`

> Ideally, I'd like a single source of truth for the list of Python supported versions (python_requires in setup.py is such a source but it's hard to parse).

true.

Reply:
> But maintainability aside, it seems a bit odd to test for our lower bound in a loop when we know precisely what the lower bound is: `sys.version_info.minor +1`

You're right.